build/gen-drivah.toml.sh

```diff
@@ -27,15 +27,35 @@ SNAPSHOT=true
 GO_TAGS=${GO_TAGS-release}
 LICENSE_PUBKEY=license.key
 
+latest_tag() {
+    local tag=$1
+    if [[ "$tag" =~ "SNAPSHOT" ]]; then
+        echo "next-SNAPSHOT"
+    else
+        echo "latest"
+    fi
+}
```

Comment:
What I did not realise until looking at this PR was that we use different repos for nightly builds and merge-xyz builds. So my argument against using `latest` for unreleased software becomes moot. 

Could we simply and just use `latest` in all cases given the we can distinguish between released software and intermediate builds by looking at `eck` vs `eck-snapshots` in the repo path?

Reply:
If we do that, we will not differentiate the two cases:
- on the nightly builds of main (nightly-main trigger):
  - `docker.elastic.co/eck-snapshots/eck-operator${suffix}:latest`
- on every merge into a release branch (merge-xyz trigger):
  -`docker.elastic.co/eck-snapshots/eck-operator${suffix}:latest`

This will be fine most of the time, but in a rare case I can imagine that the two are done concurrently and one will overwrite the image pushed by the other. We could go back to another name for the latter: docker.elastic.co/eck-ci/eck-operator-br (I think we can do whatever we want for the part three).

Reply:
Hm good point. It would be good to distinguish these. I am open wrt to the approach. 

Another thing I am wondering from the PR description:

```
# images to scan for trigger:tag-bc
docker.elastic.co/eck/eck-operator:latest
docker.elastic.co/eck/eck-operator-fips:latest
docker.elastic.co/eck/eck-operator-ubi:latest
docker.elastic.co/eck/eck-operator-ubi-fips:latest
# check drivah.toml
names = ["docker.elastic.co/eck/eck-operator"]
tags = ["1.2.3-bc1-arm64","latest-arm64"]
names = ["docker.elastic.co/eck/eck-operator"]
tags = ["1.2.3-bc1-dev-arm64","latest-arm64"]
names = ["docker.elastic.co/eck/eck-operator-fips"]
tags = ["1.2.3-bc1-arm64","latest-arm64"]
names = ["docker.elastic.co/eck/eck-operator-ubi"]
tags = ["1.2.3-bc1-arm64","latest-arm64"]
names = ["docker.elastic.co/eck/eck-operator-ubi-fips"]
tags = ["1.2.3-bc1-arm64","latest-arm64"]
```

This sounds like we would create a new `latest` tag on bc builds replacing the last released version already in the build candidate stage? Should we try to avoid that? I think we don't recommend using `latest` to our users but it might still be unexpected to have a release replaced by a BC.

Reply:
I'm looking to do:
```
nightly-main: eck-snapshots/eck-operator:latest
merge-xyz:    eck-snapshots/eck-operator:next-release-latest
tag-bc:       eck-snapshots/eck-operator:bc-latest
tag-final:    eck/eck-operator:latest
```